docs/user-guide/petset.md

```diff
@@ -0,0 +1,370 @@
+
+* It makes any startup script non-portable, users need to write specialized code for Kubernetes
+* It introduces several points of failure
+    - the secrets to talk to the API server might not be mounted in the pod
+    - the apiserver Service might be in flux
+    - kube-proxy on the node might not have updated the apiserver endpoint rules etc
+* It introduces an upgrade related failure scenario, since you're deserializing raw json from the apiserver
+* It puts load on the apiserver
+
+Pet Set gives you a way to disover your peers using DNS records. To illustrate this we can use the previous example (note: one usually doesn't `apt-get` in a container).
+
+```shell
+$ kubectl exec -it web-0 /bin/sh
+# apt-get update && apt-get install -y dnsutils
+...
+
+# nslookup -type=srv nginx.default
+Server:		10.0.0.10
+Address:	10.0.0.10#53
+
+nginx.default.svc.cluster.local	service = 10 50 0 web-1.ub.default.svc.cluster.local.
+nginx.default.svc.cluster.local	service = 10 50 0 web-0.ub.default.svc.cluster.local.
+```
+
+### Ordering
+
+While stable identites might decouple the physical infrastructure from units of compute, it's not sufficient to decouple system administration from application development. This is a problem the industry hasn't solved for many years, and we're probably not going to solve it now, but there are certain properties of an ordered set that we can leverage to significantly reduce the probability that someone wakes up to an alarming pager.
+
+As an example, consider a common application deployment setup: master/slave. This usually involves picking a single instance of a database as the "master", and parenting every other instance to it, as a "slave". The instructions for most master/slave setups begin with: contact a admin, and end with: run to the hills. Jokes aside, this is because deploying master/slave is a little delicate. The steps are:
+
+* Start 1 database as the master
+* Find its ip
+* Start all other databases pointed at the master ip
+
+We're intentionally ignoring the failover scenario.
+If you think about what the admin is doing, he/she is:
+
+1. Making sure the first database is started in stand alone mode, meaning it doesn't take any input about peers, and no peers are trying to join it.
+2. Waiting till it's "healthy" (maybe till it can serve reads)
+3. Checking its ip
+4. Starting one more database with the master ip as input
+5. Repeat from step 2 for each slave
+
+A meticulous admin would re-verify that the master ip hasn't changed on each iteration, and that all previous pets continue to believe they are slaves of the current master. The Pet Set controller tries to mimic these actions. It will always start one pet at a time and wait for it to enter Running and Ready before starting the next one. Instead of using ips, it provisions DNS records for each pet, so even if the IP does change the DNS name will not.
```

Comment:
The way you use the term "Running" and "Ready" are cool here. Incorporate them into the steps above.